src/main/java/com/endava/soa_5ed/task1_basics/Ex4.java

```diff
@@ -0,0 +1,41 @@
+package com.endava.soa_5ed.task1_basics;
+
+import java.util.Scanner;
+
+public class Ex4 {
+//    Write 3 methods that calculates sum of numbers from 0 to a given number.
+//            (1) Use for loop
+//            (2) Use while loop
+//            (3) Use do while loop
+
+    //1st method - for loop
+    int calc1(int num){
+        int sum=0;
+        for(int i=0; i<=num; i++){
+            sum+=i;
+        }
+        return sum;
+    }
+    //2nd method - while loop
+    int calc2(int num){
+        int sum=0;
+        int i=0;
+        while(i<=num){
+            sum+=i;
+            i++;
+        }
+        return sum;
+    }
+    //3rd method - do while loop
+    int calc3(int num){
+        int sum = 0;
+        int i = 0;
```

Comment:
same as above with the starting value of i

Reply:
there is a possibility that the user may provide the value 0, so if we start iterating from 1, the do{} block will add 1 to the sum, which will be wrong.

Reply:
yes, good point, I agree